File: java-client/src/main/java/com/bolour/boardgame/client/domain/PiecePoint.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PiecePoint {
    public final Piece piece;
    public final Point point;

    @JsonCreator
    public PiecePoint(
      @JsonProperty("piece") Piece piece,
      @JsonProperty("point") Point point
    ) {
        this.piece = piece;
        this.point = point;
    }
}

File: scala-server/app/com/bolour/boardgame/scala/common/domain/PiecePoint.scala
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.boardgame.scala.common.domain

import com.bolour.plane.scala.domain.Point

// TODO. Rename value to piece. Changes API. So have to do it for Haskell as well.

/**
  * Combination of a piece and its location on the board.
  */
case class PiecePoint(value: Piece, point: Point) {
  def piece = value
}
